test/test_nested_tensor_masking.py

```diff
@@ -3,42 +3,309 @@
 import pdb
 import sys
 import torch
-import nestedtensor
+import nestedtensor as NT
 import unittest
 from utils import TestCase
 import random
 
 import utils
 
 class TestTensorMask(TestCase):
-    # TODO: test comparison operator fails if nested_size doesn't agree.
+    def test_gen_nested_tensor(self):
+        nt = utils.gen_nested_tensor(seed=1, nested_dim=1, tensor_dim=1, size_low=2, size_high=2)
+        self.assertTrue(NT.is_nested_tensor(nt))
+        self.assertEqual(nt.nested_dim(), 1)
+        self.assertEqual(nt.size(), (2, 2))
+        self.assertEqual(nt[0].dim(), 1)
+        self.assertEqual(nt[0].size(), torch.Size([2]))
 
-    def test_simple(self):
-        values = [torch.rand(i) for i in range(10)]
-        random.shuffle(values)
-        nt = nestedtensor.nested_tensor(values)
-        tensor, mask = nt.to_tensor_mask()
+        nt = utils.gen_nested_tensor(seed=1, nested_dim=2, tensor_dim=1, size_low=2, size_high=2)
+        self.assertTrue(NT.is_nested_tensor(nt))
+        self.assertEqual(nt.nested_dim(), 2)
+        self.assertEqual(nt.size(), (2, 2, 2))
+        self.assertEqual(nt[0].dim(), 2)
+        self.assertEqual(nt[0].size(), (2, 2))
+        self.assertEqual(nt[0][0].size(), torch.Size([2]))
+
+        nt = utils.gen_nested_tensor(seed=1, nested_dim=2, tensor_dim=2, size_low=2, size_high=2)
+        self.assertTrue(NT.is_nested_tensor(nt))
+        self.assertEqual(nt.nested_dim(), 2)
+        self.assertEqual(nt.size(), (2, 2, 2, 2))
+        self.assertEqual(nt[0].dim(), 3)
+        self.assertEqual(nt[0].size(), (2, 2, 2))
+        self.assertEqual(nt[0][0].size(), torch.Size([2, 2]))
+        self.assertEqual(nt[0][0][0].size(), torch.Size([2]))
+
+        nt = utils.gen_nested_tensor(seed=1, nested_dim=2, tensor_dim=2, size_low=10, size_high=10)
```

Comment:
Is this actually helpful? If it works with size_low=2

Reply:
removed.